docs/architecture/self-hosted/upgrade-backup-restore.md

```diff
@@ -0,0 +1,168 @@
+| **GET** `https://{prophecy-url}/api/backup/list`   | This API returns the list of available backups.                                                                                                                                                                                                                                             | None expected |
+| **GET** `https://{prophecy-url}/api/backup/delete` | This API attempts the delete the backup data (local and upstream) and also the metadata (database entries) associated with it. Note that in case of `enableRegularBackups` set to true, backups are older than `backupRetentionCount` in reverse order are garbage collected automatically. | `timestamp`   |
+
+## Restore
+
+Restore is an on-demand based overwrite of the whole Prophecy configuration to reflect the state of a particular backup.
+The restore operation always assumes a running destination Prophecy cluster where the data and the configuration of source cluster will be restored.
+
+:::note
+If backup was taken in Athena's local Persistent Volume, it needs to be copied to Athena's Persistent Volume in the destination cluster before the restore operation can be performed.
+:::
+
+### On-demand restore
+
+You can restore using the Restore API. See [Generate API Key](./generate-api-key/) if you need an API key.
+
+:::danger
+This API should be used with extreme caution as triggering this will lead to loss of current state/data.
+:::
+
+The below API is used to trigger a restore operation. It expects one parameter which is the `timestamp` of a successful backup.
+
+```
+curl --location --request POST 'https://{prophecy-url}/api/restore' \
+--header 'Cookie: prophecy-token={api-key}' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "timestamp": "2022-11-22t10-00-00",
+    "sourceNamespace": "{Source cluster controlplane namespace}"
+}'
+```
+
+Sample API call with disable of gitserver restore. You may use similar options for `artifactory` / `edweb` / `metagraph` / `openidfederator`.
+
+```
+curl --location --request POST 'https://{prophecy-url}/api/restore' \
+--header 'Cookie: prophecy-token={api-key}' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "timestamp": "2022-11-22t10-00-00",
+    "sourceNamespace": "{Source cluster controlplane namespace}",
+    "svcs": {
+        "gitserver": {
+            "disable": false
+        },
+    }
+}'
+```
+
+In the above API:
+
+- `timestamp` is the timestamp of the backup to use to perform the restore
+- `sourceNamespace` is the namespace in which source cluster's control plane was installed.
+- `svcs`: This JSON object needs to be set only when you wish to skip restore of any particular service by setting `disabled` as `true`. Otherwise, you can skip `svcs` field entirely.
+
+Sample response
+
+```
+{
+    "code": 202,
+    "message": "Request Accepted"
+}
+```
+
+### Additional restore APIs
+
```

Comment:
I think we should provide at least one sample call format like

 curl --location --request POST 'https://{prophecy-url}/api/backup' \
 --header 'Cookie: prophecy-token={api-key}' \
 --header 'Content-Type: application/json' \
 --data-raw '{}'
So as to know the format in which the call is to be invoked. The token here is an important one.